concepts/function-arguments/about.md

```diff
@@ -1,2 +1,231 @@
+
+Following is an example of keyword arguments being referred to by their parameter name and by position:
+
+```python
+>>> def concat(greeting, name):
+...         return f"{greeting}{name}"
+... 
+# Function call using parameter names as argument keywords.
+>>> print(concat(name="Bob", greeting="Hello, "))
+Hello, Bob
+...
+# Function call with positional data as arguments.
+>>> print(concat("Hello, ", "Bob"))
+Hello, Bob
+>>> print(concat())
+Hello, you
+
+```
+
+Requiring keyword-only arguments for function calls can be done through the use of the `*` operator in the parameter list:
+
+Following is an example of keyword-only arguments:
+
+```python
+# Function definition requiring keyword-only arguments.
+>>> def concat(*, greeting, name):
+...         return f"{greeting}{name}"
+... 
+# Keyword arguments can be in an arbitrary order.
+>>> print(concat(name="Bob", greeting="Hello, "))
+Hello, Bob
+...
+# Calling the function with positional data raises an error.
+>>> print(concat("Hello, ", "Bob"))
+Traceback (most recent call last):
+    print(concat("Hello, ", "Bob"))
+TypeError: concat() takes 0 positional arguments but 2 were given
+
+
+```
+
+## Positional or Keyword Arguments
+
+Arguments can be positional or keyword if neither the `/` nor `*` operators are used in the parameter definitions.
+Alternately, the positional-or-keyword arguments can be placed between the positional-only parameters on the left and the keyword-only parameters on the right.
+
+Following is an example of positional-only, positional-or-keyword, and keyword-only arguments:
+
+```python
+# Position-only argument followed by position-or-keyword, followed by keyword-only.
+>>> def concat(greeting, /,  name, *, ending):
+...         return f"{greeting}{name}{ending}"
+... 
+>>> print(concat("Hello, ", "Bob", ending="!"))
+Hello, Bob!
+>>> print(concat("Hello, ", name="Bob", ending="!"))
+Hello, Bob!
+...
+>>> print(concat(greeting="Hello, ", name="Bob", ending="!"))
+Traceback (most recent call last):
+    print(concat(greeting="Hello, ", name="Bob", ending="!"))
+TypeError: concat() got some positional-only arguments passed as keyword arguments: 'greeting'
+
+
+## `*args`
+
+Code examples will often use a function definition something like the following:
+
+```python
+def my_function(*args, **kwargs):
+    # code snipped
+
+```
+
+`*args` is a two-part name that represents a `tuple` with an indefinite number of separate positional arguments, also known as a [`variadic argument`][variadic argument].
+
+`args` is the name given to the `tuple` of arguments,  but it could be any other valid Python name, such as `my_args`, `arguments`, etc.
+The `*` is the operator which transforms the group of separate arguments into a [`tuple`][tuple].
+Since a tuple can be iterated, `args` can be passed to any other function which takes an iterable.
+Although `*args` is commonly juxtaposed with `**kwargs`, it doesn't have to be.

```

Comment:
Flagging this again because I think it was missed.  This example is problematic because it confuses tuple unpacking with the functionality of `*args`.  While both use an overloaded `*` operator, one _makes_ a tuple from a group of arguments, the other _unpacks_ a data structure into multiple individual variables or arguments.  It is very confusing for students at first, so we really need to separate out the two behaviors.  We need to especially avoid implying that tuple unpacking for function calls needs the `*arg` parameter -- it doesn't.